src/modules/Core/Account/settings.js

```diff
@@ -95,18 +95,40 @@ export const setDenominationKeyRequest = (account: AbcAccount, currencyCode: str
 // Helper Functions
 export const getSyncedSettings = (account: AbcAccount) =>
   getSyncedSettingsFile(account).getText()
-  .then(JSON.parse)
+  .then((text) => {
+    const settingsFromFile = JSON.parse(text)
+    return settingsFromFile
+  })
   .catch((e) => {
     console.log(e)
     // If Settings.json doesn't exist yet, create it, and return it
     return setSyncedSettings(account, SYNCED_ACCOUNT_DEFAULTS)
   })
 
+export async function getSyncedSettingsAsync (account: AbcAccount) {
+  try {
+    const file = await getSyncedSettingsFile(account)
+    const text = await file.getText()
+    const settingsFromFile = JSON.parse(text)
+    return settingsFromFile
+  } catch (e) {
+    console.log(e)
+    // If Settings.json doesn't exist yet, create it, and return it
+    return setSyncedSettings(account, SYNCED_ACCOUNT_DEFAULTS)
+  }
+}
+
 export const setSyncedSettings = (account: AbcAccount, settings: Object) => {
   const text = JSON.stringify(settings)
   const SettingsFile = getSyncedSettingsFile(account)
+  SettingsFile.setText(text)
+  .then(() => settings)
```

Comment:
What is this line for? 

The `.then(() => settings)` won't do anything. Are you trying to return the settings to the caller of setSyncedSettings?